apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/Logging/ILoggerExtensions.cs

```diff
@@ -0,0 +1,98 @@
+﻿// Copyright 2021 Google LLC
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     https://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Api.Gax;
+using Google.Cloud.Logging.V2;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Google.Cloud.Diagnostics.Common
+{
+    /// <summary>
+    /// Extensions for <see cref="ILogger"/> to allow augmenting the logged information with contextual data.
+    /// GoogleLogger(TODO: change this to a cref when I've moved it to this package)
+    /// will add the information to the <see cref="LogEntry"/> that is sent to Google Cloud Logging.
+    /// For the consumption of other loggers, the information will be included in newly created scopes
+    /// at the moment of logging, in a format that all loggers should understand.
+    /// </summary>
+    public static class ILoggerExtensions
+    {
+        /// <summary>
```

Comment:
Should we have overloads for these methods accepting `IEnumerable<KeyValuePair<string, string>>` as well? That way users can supply a `Dictionary<string, string>` for example.

Reply:
Done